### PR TITLE
Control video in fullscreen mode using arrows and numbers

### LINKED ISSUE #6606: [Feature Request] Video player skip some seconds when right/left arrow keys are pressed
### Is your feature request related to a problem?
No.

### Describe the solution you'd like
While playing any video in the __Media Viewer__ I'd like to be able to skip 5/10 or any amount near those of seconds by pressing the right key, and to rewind the same amount if pressing the left key.
Also the other two arrows could be used to navigate between videos or adjust the volume.

### Describe alternatives you've considered
Any other hotkey could be used to achieve the same thing, but it wouldn't be as user friendly as the the arrow keys are.

### Additional context
If needed, ask and I'll provide some, but I don't think they are necessary.

### PR DESCRIPTION
**Pull request by Hermesiss**
Closes #6606

### CODE REVIEW

**Comment by Aokromes** [comment]
i don't think shortcuts doing different things between various parts of application is wise.

**Comment by Hermesiss** [comment]
I can move it to a proper place. Number keys are processed in several places so I decided to put them with other media shortcuts

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 1]
```diff
@@ -3002,6 +3002,18 @@ void OverlayWidget::playbackPauseResume() {
 	}
 }
 
+void OverlayWidget::seekRelativeTime(crl::time time) {
+	Expects(_streamed != nullptr);
```
Comment (john-preston) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3006]: Please leave a newline after preconditions (`Expects(...)`)

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 2]
```diff
@@ -3002,6 +3002,18 @@ void OverlayWidget::playbackPauseResume() {
...
+	crl::time newTime = std::clamp(_streamed->instance.info().video.state.position + time, 0LL, _streamed->instance.info().video.state.duration);
```
Comment (john-preston) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3007]: Please use `const auto ...` variable declaration where appropriate.

Also here the line length is too long, the rest of the codebase follows such style in such cases:
```cpp
const auto newTime = std::clamp(
    _streamed->instance.info().video.state.position + time,
    crl::time(0),
    _streamed->instance.info().video.state.duration);
```

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 3]
```diff
@@ -3002,6 +3002,18 @@ void OverlayWidget::playbackPauseResume() {
...
+	restartAtSeekPosition(newTime);
+}
+
+void OverlayWidget::restartAtProgress(float64 progress) {
+	Expects(_streamed != nullptr);
```
Comment (john-preston) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3012]: Please leave a newline after preconditions (Expects(...))

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 4]
```diff
@@ -3738,6 +3750,24 @@ void OverlayWidget::handleKeyPress(not_null<QKeyEvent*> e) {
 			if (key == Qt::Key_Escape) {
 				playbackToggleFullScreen();
 			}
+			else if (key == Qt::Key_0) {
```
Comment (john-preston) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3753]: Please use code style consistent with other places, specifically here - place `else if` on the same line as the closing brace, like:
```cpp
} else if (...) {
    ...
} else if (...) {
    ...
}
```

**Review by john-preston — CHANGES_REQUESTED** [review]
LGTM except some minor style changes.

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 5]
```diff
@@ -3737,7 +3751,21 @@ void OverlayWidget::handleKeyPress(not_null<QKeyEvent*> e) {
...
+			} else if (key >= Qt::Key_1 && key <= Qt::Key_9) {
+				activateControls();
+				const auto index = int(key - Qt::Key_0);
+				restartAtProgress(index / 10.0);
+			} else if (key == Qt::Key_Left) {
+				activateControls();
+				seekRelativeTime(-5000);
+			} else if (key == Qt::Key_Right) {
+				activateControls();
+				seekRelativeTime(5000);
```
Comment (23rd) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3766]: 5000 ms can be moved to `constexpr auto`.

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 6]
```diff
@@ -3002,6 +3002,20 @@ void OverlayWidget::playbackPauseResume() {
 	}
 }
 
+void OverlayWidget::seekRelativeTime(crl::time time) {
+	Expects(_streamed != nullptr);
+
+	const auto newTime = std::clamp(_streamed->instance.info().video.state.position + time, 0LL, _streamed->instance.info().video.state.duration);
```
Comment (23rd) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3008]: This line is still too long.

**Comment by 23rd** [comment]
By the way style changes can be squashed with the first commit.

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 7]
```diff
@@ -3002,6 +3003,22 @@ void OverlayWidget::playbackPauseResume() {
 	}
 }
 
+void OverlayWidget::seekRelativeTime(crl::time time) {
+	Expects(_streamed != nullptr);
+
+	const auto newTime = std::clamp(_streamed->instance.info().video.state.position + time,
```
Comment (23rd) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3009]:
```suggestion
	const auto newTime = std::clamp(
		_streamed->instance.info().video.state.position + time, 
```

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 8]
```diff
@@ -94,6 +94,7 @@ constexpr auto kPreloadCount = 3;
 constexpr auto kMaxZoomLevel = 7; // x8
 constexpr auto kZoomToScreenLevel = 1024;
 constexpr auto kOverlayLoaderPriority = 2;
+constexpr auto kSeekTimeMs = 5000;
```
Comment (john-preston) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:97]: The last small thing - please use 5 * crl::time(1000) here, that way it has the correct type with the format like in the rest of the codebase. Thanks.

**Review by john-preston — CHANGES_REQUESTED** [review]
I think that's all :) Thank you.

**Review by john-preston — APPROVED** [review]
Ok, I've started the CI-s, when they build without problems I'll merge it for the next beta version.

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 9]
```diff
@@ -3002,6 +3003,23 @@ void OverlayWidget::playbackPauseResume() {
...
+		_streamed->instance.info().video.state.position + time,
+		crl::time(0),
+		_streamed->instance.info().video.state.duration);
+	restartAtSeekPosition(newTime);
+}
+
+void OverlayWidget::restartAtProgress(float64 progress) {
+	Expects(_streamed != nullptr);
+
+	progress = std::clamp(progress, 0., 1.);
```
Comment (23rd) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3019]:
```suggestion
	progress = std::clamp(progress, 0., 1.);
```

File: Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp [hunk 10]
```diff
@@ -3002,6 +3003,23 @@ void OverlayWidget::playbackPauseResume() {
...
+		crl::time(0),
+		_streamed->instance.info().video.state.duration);
+	restartAtSeekPosition(newTime);
+}
+
+void OverlayWidget::restartAtProgress(float64 progress) {
+	Expects(_streamed != nullptr);
+
+	progress = std::clamp(progress, 0., 1.);	
+	restartAtSeekPosition(_streamed->instance.info().video.state.duration * progress);
```
Comment (23rd) [inline Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp:3020]:
```suggestion
	restartAtSeekPosition(_streamed->instance.info().video.state.duration
		* std::clamp(progress, 0., 1.));
```

**Comment by Hermesiss** [comment]
Please restart the CI (or see the previous [successful](https://github.com/telegramdesktop/tdesktop/actions?query=branch%3Avideo-keyboard-controls))

**Comment by john-preston** [comment]
Thanks!

**Pull request merged**